SurveyAppTests/Tests/Services/Network/Services/AuthenticationServiceSpec.swift

```diff
@@ -0,0 +1,97 @@
+// swiftlint:disable all
+//
+//  AuthenticationServiceSpec.swift
+//  SurveyAppTests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+import OHHTTPStubs
+
+@testable import SurveyApp
+
+final class AuthenticationServiceSpec: QuickSpec {
+
+    override func spec() {
+
+        var authenticationService: AuthenticationService!
+        var userSession: UserSessionProtocolMock!
+        var api: BaseAPI!
+
+        describe("AuthenticationService") {
+
+            afterEach {
+                HTTPStubs.removeAllStubs()
+            }
+
+            beforeEach {
+                userSession = UserSessionProtocolMock()
+                api = BaseAPI(userSession: userSession)
+
+                authenticationService = AuthenticationService(api: api, baseURL: Defines.baseURL)
+            }
+
+            // MARK: – authenticateEmail
+            context("when call authenticateEmail") {
+
+                let grantType = "password"
+                let email = "email@example.com"
+                let password = "12345678"
+                let clientId = "id"
+                let clientSecret = "secret"
+
+                let successRequestBody = [
+                    "grant_type": grantType,
+                    "email": email,
+                    "password": password,
+                    "client_id": clientId,
+                    "client_secret": clientSecret
+                ]
+
+                it("and get success response should receive response object correctly") {
+                    let data = JSON.AuthenticationService.authenticateEmailSuccess
+
+                    stub(condition: pathEndsWith("/api/v1/oauth/token") && hasJsonBody(successRequestBody)) { _ in
```

Comment:
if you create an enum with following capability, it will be easier to reuse for Network layer and testing.

```

enum EndpointType {

    case authExchange
    case contractsQuotations(id: String)

    var baseURL: String {
        if let baseURL = LaunchEnvironment.stringFor(.mockAPIURLKey) {
            return baseURL
        }
        switch self {
        default:
            return Constants.Network.baseURLPath
        }
    }
    
    var endpoint: String {
        switch self {
        case .authExchange: return "auth/exchange"
        case .contractsQuotations(let id): return "contracts/\(id)/quotations"
        }
    }
}
```

then in your `NetworkService` you can change url(forEndPoint...) to 
```
    func url(forEndpoint type: EndpointType) -> String {
        api.url(forEndpoint: type.endpoint, baseURL: type.baseURL)
    }
```